runtime/random.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ * Copyright (c) 2018 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+abstract class RNG {
+    abstract next() : number;
+}
+
+/**
+ * A basic random number generator using Math.random();
+ */
+class MathRandomRNG extends RNG {
+    next() : number {
+        return Math.random();
+    }
+}
+
+/**
+ * Provides a deterministic Random Number Generator for Tests
+ */
+class SeededRNG extends RNG {
+    private seed: number = 0;
+    next(): number {
+        this.seed = Math.pow(this.seed + Math.E, Math.PI) % 1;
+        return this.seed;
+    }
+}
+
+export namespace Random {
```

Comment:
I saw some advice not to mix namespaces and modules, as modules provide their own namespaces.

I think, based on this advice, that namespaces are mainly for when you want to mix lots of functionality into a single module, e.g. the 'ambient' modules.

On the other hand, the approach here nicely provides a static interface without requiring a 'class' wrapped around it, so maybe this is in fact appropriate :)

Maybe worth discussing with @shaper to see what he thinks?

Reply:
Will hold off on this until the typescript situation stabilizes...